src/phoenix/datasets/types.py

```diff
@@ -183,3 +208,202 @@ def __post_init__(self) -> None:
+        raise NotImplementedError
+
+    @classmethod
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        # Direct sub-classing by users is discouraged.
+        raise NotImplementedError
+
+
+@dataclass(frozen=True)
+class TaskSummary(_HasStats):
+    _title: str = "Experiment Summary"
+
+    @classmethod
+    def from_task_runs(
+        cls,
+        config: ExperimentConfig,
+        task_runs: Iterable[Optional[ExperimentRun]],
+    ) -> "TaskSummary":
+        df = pd.DataFrame.from_records(
+            [
+                {
+                    "example_id": run.dataset_example_id,
+                    "error": bool(run.error),
+                }
+                for run in task_runs
+                if run is not None
+            ]
+        )
+        n_runs, n_errors = len(df), 0 if df.empty else df["error"].sum()
+        record = {
+            "n": config.count,
+            "n_runs": n_runs,
+            "n_errors": n_errors,
+            "pct_errors": n_errors / n_runs,
+        }
+        stats = pd.DataFrame.from_records([record])
+        summary: TaskSummary = object.__new__(cls)
+        summary.__init__(stats=stats)  # type: ignore[misc]
+        return summary
+
+    @classmethod
+    def __new__(cls, *args: Any, **kwargs: Any) -> Any:
+        # Direct instantiation by users is discouraged.
+        raise NotImplementedError
+
+    @classmethod
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        # Direct sub-classing by users is discouraged.
+        raise NotImplementedError
+
+
+@dataclass(frozen=True)
+class RanExperiment(Experiment):
+    config: ExperimentConfig
+    dataset: Dataset = field(repr=False)
+    runs: Sequence[ExperimentRun] = field(repr=False)
+    task_summary: TaskSummary = field(repr=False)
+    eval_summaries: Tuple[EvaluationSummary, ...] = field(repr=False, default=())
+
+    @property
+    def url(self) -> str:
+        return get_experiment_url(dataset_id=self.dataset.id, experiment_id=self.id)
+
+    @property
+    def info(self) -> str:
+        return f"🔗 View this experiment: {self.url}"
+
+    def __add__(self, eval_summary: EvaluationSummary) -> "RanExperiment":
+        eval_summaries = (eval_summary, *self.eval_summaries)
+        ran_experiment: RanExperiment = object.__new__(RanExperiment)
+        ran_experiment.__init__(  # type: ignore[misc]
+            **{**_asdict(self), "eval_summaries": eval_summaries}
+        )
+        return ran_experiment
+
+    def __str__(self) -> str:
+        summaries = (self.task_summary, *self.eval_summaries)
+        return f"\n{self.info}\n\n" + "\n\n".join(map(str, summaries))
+
+    @classmethod
+    def __new__(cls, *args: Any, **kwargs: Any) -> Any:
+        # Direct instantiation by users is discouraged.
+        raise NotImplementedError
+
+    @classmethod
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        # Direct sub-classing by users is discouraged.
+        raise NotImplementedError
```

Comment:
This seems unnecessary if we don't document or advertise to users.

Reply:
yea, we can take it out later. just want to be super duper extra safe